pkg/skaffold/build/kaniko.go

```diff
@@ -55,28 +56,34 @@ func (k *KanikoBuilder) Build(ctx context.Context, out io.Writer, tagger tag.Tag
 		return nil, errors.Wrap(err, "getting kubernetes client")
 	}
 
-	secretData, err := ioutil.ReadFile(k.KanikoBuild.PullSecret)
-	if err != nil {
-		return nil, errors.Wrap(err, "reading secret")
-	}
+	if k.KanikoBuild.PullSecret == "" {
+		logrus.Debug("No pull secret specified. Checking for one in the cluster.")
+		if _, err := client.CoreV1().Secrets(k.KanikoBuild.Namespace).Get(k.KanikoBuild.PullSecretName, metav1.GetOptions{}); err != nil {
+			return nil, errors.Wrap(err, "checking for existing kaniko secret")
+		}
+	} else {
+		secretData, err := ioutil.ReadFile(k.KanikoBuild.PullSecret)
+		if err != nil {
+			return nil, errors.Wrap(err, "reading secret")
+		}
 
-	_, err = client.CoreV1().Secrets("default").Create(&v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   "kaniko-secret",
-			Labels: map[string]string{"kaniko": "kaniko"},
-		},
-		Data: map[string][]byte{
-			"kaniko-secret": secretData,
-		},
-	})
-	if err != nil {
-		logrus.Warnf("creating secret: %s", err)
-	}
-	defer func() {
-		if err := client.CoreV1().Secrets("default").Delete("kaniko-secret", &metav1.DeleteOptions{}); err != nil {
-			logrus.Warnf("deleting secret")
+		if _, err := client.CoreV1().Secrets(k.KanikoBuild.Namespace).Create(&v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   k.KanikoBuild.PullSecretName,
+				Labels: map[string]string{"skaffold-kaniko": "skaffold-kaniko"},
+			},
+			Data: map[string][]byte{
+				constants.DefaultKanikoSecretName: secretData,
+			},
+		}); err != nil {
+			logrus.Warnf("creating secret: %s", err)
```

Comment:
I guess these were there before but should we really just warn if we fail to create the secret? This should most likely throw an error.